index.js

```diff
@@ -0,0 +1,259 @@
+const navWrapper = document.querySelector(".nav-wrapper");
+const navToggle = document.querySelector(".mobile-nav-toggle");
+const primaryNavigation = document.querySelector(".primary-navigation");
+
+document.body.addEventListener("click", (e) => {
+  if (
+    !navWrapper.contains(e.target) &&
+    navToggle.attributes[2].nodeValue === "true"
+  ) {
+    closeNav();
+  }
+});
+navWrapper.addEventListener("click", (e) => {
+  if (navToggle.attributes[2].nodeValue === "false") {
+    showNav();
+  } else {
+    closeNav();
+  }
+});
+
+const showNav = () => {
+  navToggle.setAttribute("aria-expanded", true);
+  primaryNavigation.toggleAttribute("opening");
+  primaryNavigation.addEventListener(
+    "animationend",
+    () => {
+      primaryNavigation.toggleAttribute("opened");
+      primaryNavigation.toggleAttribute("opening");
+    },
+    { once: true }
+  );
+};
+
+const closeNav = () => {
+  primaryNavigation.toggleAttribute("closing");
+  navToggle.setAttribute("aria-expanded", false);
+  primaryNavigation.addEventListener(
+    "animationend",
+    () => {
+      primaryNavigation.toggleAttribute("opened");
+      primaryNavigation.toggleAttribute("closing");
+    },
+    { once: true }
+  );
+};
+
+const contactMeButton = document.querySelector("#contact-me");
+const modal = document.querySelector("#modal");
+const closeModalButton = document.querySelector(".modal__close");
+const mobileContactButton = document.querySelector(".button--mobile");
+
+const closeModalAnimation = () => {
+  modal.setAttribute("closing", "");
+
+  modal.addEventListener(
+    "animationend",
+    () => {
+      modal.removeAttribute("closing");
+      modal.close();
+    },
+    { once: true }
+  );
+  document.body.style.overflow = "auto";
+};
+
+contactMeButton.addEventListener("click", () => {
+  modal.showModal();
+  document.body.style.overflow = "hidden";
+});
+
+mobileContactButton.addEventListener("click", () => {
+  modal.showModal();
+});
+
+modal.addEventListener("click", (e) => {
+  e.stopPropagation();
+  if (
+    e.target === modal ||
+    e.target.type === "submit" ||
+    e.target === closeModalButton
+  ) {
+    closeModalAnimation();
+    modal.close();
+  }
+});
+
+/* This isn't consistant enough for me .... I'm not sure if it is just breaking when live server or fireFox disable's scroll anchoring due to many calls when resizing too frequently or I'm just doing something wrong.
```

Comment:
I don't know exactly what this bug looks like, but my knee jerk reaction is to say don't worry about it. I think you're using basic DOM to solve pretty touch positioning challenges, where you're having to fight layout. Looking at this feature I'd think you'd use `<canvas>`, which I would say is too advanced for this course.

Reply:
I found a js library called popper.js that has this feature. Are importing and utilizing libraries in a junior portfolio looked down upon? I ask this because the overall sentiment on the web is everything on the portfolio should be made by you, and you should not utilize things like templates or css libraries/frameworks.

Reply:
To answer your question, importing libraries is not looked down upon. Further along in this course you'll learn how to do this.